addons/sale/controllers/portal.py

```diff
@@ -183,6 +183,10 @@ def portal_order_page(self, order_id, report_type=None, access_token=None, messa
             values['pms'] = request.env['payment.token'].search(
                 [('partner_id', '=', order_sudo.partner_id.id),
                 ('acquirer_id', 'in', acquirers.filtered(lambda acq: acq.payment_flow == 's2s').ids)])
+        else:
+            tx = order_sudo.get_portal_last_transaction()
+            if tx and tx.state == 'done':
```

Comment:
```suggestion
            if tx and tx.state == 'done' and tx.is_processed:
```

I suggest to check first that the payment has been processed

Reply:
updated